### PR TITLE
EP-1047 - Adding Amount to Pay Buttton

### PR DESCRIPTION
**Pull request by AlexeiJovmir**
add shouldPaymentButonDisplayAmount in config object
rename 'shouldDisplayAmount' -> `shouldPaymentMethodsDisplayAmount`
generate pay button title in JPTransactionInteractorImpl
update tests

### CODE REVIEW

File: Source/Modules/Transaction/Interactor/JPTransactionInteractor.m [hunk 1]
```diff
@@ -99,6 +100,13 @@ - (void)handleCameraPermissionsWithCompletion:(void (^)(AVAuthorizationStatus))c
     }
 }
 
+- (NSString *)generatePayButtonTitle {
+    if (self.configuration.uiConfiguration.shouldPaymentButonDisplayAmount) {
+        return [NSString stringWithFormat:@"%@ %@%@", @"pay".localized, self.configuration.amount.currency.toCurrencySymbol, self.configuration.amount.amount];
```
Comment (stpopa) [inline Source/Modules/Transaction/Interactor/JPTransactionInteractor.m:105]: to make this expression shorter, let's extract `self.configuration.amount ` to local variable

File: Source/Models/UIConfiguration/JPUIConfiguration.h [hunk 2]
```diff
@@ -31,7 +31,12 @@
 /**
  * A boolean property that defines if the amount should be displayed on the payment method screen
  */
-@property (nonatomic, assign) BOOL shouldDisplayAmount;
+@property (nonatomic, assign) BOOL shouldPaymentMethodsDisplayAmount;
+
+/**
+ * A boolean property that defines if the amount should be displayed on the card pay screen
```
Comment (stpopa) [inline Source/Models/UIConfiguration/JPUIConfiguration.h:37]: `payment methods only`

File: Examples/ObjectiveCExampleApp/ObjectiveCExampleApp/Models/Settings.m [hunk 3]
```diff
@@ -119,8 +119,12 @@ - (BOOL)isAVSEnabled {
     return [self.defaults boolForKey:kAVSEnabledKey];
 }
 
-- (BOOL)isAmountLabelEnabled {
-    return [self.defaults boolForKey:kAmountLabelEnabledKey];
+- (BOOL)isPaymentMethodsDisplayAmount {
+    return [self.defaults boolForKey:kPaymentMethodsDisplayAmount];
+}
+
+- (BOOL)isPaymentButonDisplayAmount {
```
Comment (stpopa) [inline Examples/ObjectiveCExampleApp/ObjectiveCExampleApp/Models/Settings.m:126]: `shouldPaymentButonDisplayAmount `

File: Examples/ObjectiveCExampleApp/ObjectiveCExampleApp/Models/Settings.m [hunk 4]
```diff
@@ -119,8 +119,12 @@ - (BOOL)isAVSEnabled {
     return [self.defaults boolForKey:kAVSEnabledKey];
 }
 
-- (BOOL)isAmountLabelEnabled {
-    return [self.defaults boolForKey:kAmountLabelEnabledKey];
+- (BOOL)isPaymentMethodsDisplayAmount {
```
Comment (stpopa) [inline Examples/ObjectiveCExampleApp/ObjectiveCExampleApp/Models/Settings.m:122]: `shouldPaymentMethodsPaymentButonDisplayAmount`

File: Examples/ObjectiveCExampleApp/ObjectiveCExampleApp/Models/Settings.h [hunk 5]
```diff
@@ -41,7 +41,8 @@ static NSString * const kApplePayPaymentMethodEnabledKey = @"is_payment_method_a
 #pragma mark - Others section keys
 
 static NSString * const kAVSEnabledKey = @"is_avs_enabled";
-static NSString * const kAmountLabelEnabledKey = @"is_amount_label_enabled";
+static NSString * const kPaymentMethodsDisplayAmount = @"is_amount_method_enabled";
+static NSString * const kPaymentButonDisplayAmount = @"is_amount_button_enabled";
```
Comment (stpopa) [inline Examples/ObjectiveCExampleApp/ObjectiveCExampleApp/Models/Settings.h:45]: `kShouldPaymentMethodsDisplayAmount = @"should_payment_methods_display_amount"'`
`kShouldPaymentButonDisplayAmount = @"should_payment_buton_display_amount"'`

File: Source/Models/UIConfiguration/JPUIConfiguration.h (hunk 2 above)
Comment (AlexeiJovmir) [reply Source/Models/UIConfiguration/JPUIConfiguration.h:37]: Screen Shot 2020-06-09 at 13 48 51
something like?

**Pull request merged**